### ISSUE TITLE
forceColorScheme lead to error when changing page on Next.js

### ISSUE THREAD

**Issue by ramiel** [mantinedev/mantine#4959]
### What package has an issue

@mantine/core

### Describe the bug

My provider is set as follow

```tsx
<MantineProvider
      theme={theme}
      forceColorScheme={isHomePage ? "dark" : undefined}
      defaultColorScheme="auto"
>
```

this way I force the colorscheme to be `dark` in the homepage. Now, if I navigate away from homepage (without reloading, client side navigation), the next page has something wrong. Some components follow the dark scheme, others are in light. If I load that page directly, without going through homepage, it works correctly.

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.1.1

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

### Possible fix

Maybe, if the `forcecolorscheme` changes from some value to undefined, the colorscheme must fallback to default (or user set one)

**Comment by rtivital** [comment]
Please provide a repository with a reproduction and steps to reproduce the issue.

**Comment by ramiel** [comment]
Sure, here it is: https://github.com/ramiel/mantine-forcecolorscheme-issue

Visit the home page and then click on the link. Look at the same page if loaded directly.

**Comment by ramiel** [comment]
In case you want to check the deployed version: https://mantine-forcecolorscheme-issue.vercel.app/

**Comment by rtivital** [comment]
Fixed in [7.4.2](https://github.com/mantinedev/mantine/releases/tag/7.4.2)
